plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryConnectorModule.java

```diff
@@ -166,10 +167,19 @@ protected void setup(Binder binder)
                     .to(IdentityCacheMapping.SingletonIdentityCacheMapping.class)
                     .in(Scopes.SINGLETON);
 
-            newOptionalBinder(binder, BigQueryCredentialsSupplier.class)
+            OptionalBinder<BigQueryCredentialsSupplier> credentialsSupplierBinder = newOptionalBinder(binder, BigQueryCredentialsSupplier.class);
+            credentialsSupplierBinder
                     .setDefault()
-                    .to(StaticBigQueryCredentialsSupplier.class)
+                    .to(DefaultBigQueryCredentialsProvider.class)
                     .in(Scopes.SINGLETON);
+
+            StaticCredentialsConfig staticCredentialsConfig = buildConfigObject(StaticCredentialsConfig.class);
+            if (staticCredentialsConfig.getCredentialsFile().isPresent() || staticCredentialsConfig.getCredentialsKey().isPresent()) {
+                credentialsSupplierBinder
+                        .setBinding()
+                        .to(StaticBigQueryCredentialsSupplier.class)
```

Comment:
static one also needs the proxied transport. The service account key still requires to go talk to oauth2.googleapis.com to get an access token, the service account key itself is not an access token.

Reply:
Makes sense. Thanks for investigating that

Reply:
Done. I can indeed see extra proxied calls